src/components/solvers/TextInputMask.tsx

```diff
@@ -0,0 +1,66 @@
+import { Divider, Text, Textarea } from "@chakra-ui/react";
+import Head from "next/head";
+import React, { ChangeEvent, ReactElement, useState } from "react";
+import { Container } from "../Container";
+import { Main } from "../Main";
+import { InputButtonPanel } from "./buttons/InputButtonPanel";
+import { Help } from "./SAT/Help";
+import { SolverTitle } from "./SolverTitle";
+
+export interface TextInputMaskProperties {
+    title: string;
+    description: string;
+    textPlaceholder: string;
+    onTextChanged: (text: string) => void;
+    body?: ReactElement;
```

Comment:
Isn't this supposed to contain a preview / rendering of the text data? In that case, I think that `preview` would be a more descriptive name

Reply:
The body usually contains the ProgressHandler as well as anything else you might want like a GraphArea. So it's not really a preview.